File: node/src/models/Response.js
const mongoose = require('../config/moogose');

const responseSchema = new mongoose.Schema({
  id_user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  id_link: { type: String, required: true, unique: true },
  id_personality: { type: mongoose.Schema.Types.ObjectId, ref: 'Personality', required: false },
  statistique: { type: String, required: false },
  status: { type: String, required: true },
  date: { type: String, required: true },
  content: { type: String, required: false },
});

const Response = mongoose.model('Response', responseSchema);

module.exports = Response;

File: backend/schemas/response.py
# schemas/response.py

from pydantic import BaseModel, Field
from datetime import datetime
from typing import Dict, Optional, Union
from .user import UserResponse

class ResponseBase(BaseModel):
    user_id: UserResponse
    id_link: Optional[str] = None
    status: Optional[str] = None
    id_personality: Optional[str] = None
    statistique: Optional[Dict[str, Union[str, None]]] = None
    content: Optional[Dict[str, Union[bool, None]]] = None
    date: datetime

class ResponseCreate(BaseModel):
    user_id: str 
    
class ResponseUpdate(ResponseBase):
    pass

class ResponseOut(ResponseBase):
    id: str = Field(..., alias='id', example="60d5f446f1b8e6c7b4efddf3")

    class Config:
        orm_mode = True
        from_attributes = True


